svm/src/message_processor.rs

```diff
@@ -90,7 +91,28 @@ pub(crate) fn process_message(
             .total_us += process_instruction_us;
 
         result.map_err(|err| {
-            TransactionError::InstructionError(top_level_instruction_index as u8, err)
+            let error_attribution = invoke_context.get_first_error_attribution();
+            let (inner_instruction_index, responsible_program_address) = match error_attribution {
+                Some(attr) => (
+                    attr.inner_instruction_index.map(|i| i as u8),
+                    Some(attr.responsible_program_address),
+                ),
+                None => {
```

Comment:
What I _wanted_ to do is to make this an invariant; you should never be able to reach this code without something in the SVM having blamed a program for the error. I don't see a way to provably make this so, so instead I've opted for a ‘soft fail’ that sets the attribution data to `None`. cc/ @Lichtso.